packages/@n8n/task-runner/src/js-task-runner/__tests__/js-task-runner.test.ts

```diff
@@ -396,6 +396,177 @@ describe('JsTaskRunner', () => {
 			});
 		});
 
+		describe("$getWorkflowStaticData('global')", () => {
+			it('should have the global workflow static data available in runOnceForAllItems', async () => {
+				const outcome = await execTaskWithParams({
+					task: newTaskWithSettings({
+						code: 'return { val: $getWorkflowStaticData("global") }',
+						nodeMode: 'runOnceForAllItems',
+					}),
+					taskData: newDataRequestResponse(inputItems.map(wrapIntoJson), {
+						staticData: {
+							global: { key: 'value' },
+						},
+					}),
+				});
+
+				expect(outcome.result).toEqual([wrapIntoJson({ val: { key: 'value' } })]);
+			});
+
+			it('should have the global workflow static data available in runOnceForEachItem', async () => {
+				const outcome = await execTaskWithParams({
+					task: newTaskWithSettings({
+						code: 'return { val: $getWorkflowStaticData("global") }',
+						nodeMode: 'runOnceForEachItem',
+					}),
+					taskData: newDataRequestResponse(inputItems.map(wrapIntoJson), {
+						staticData: {
+							global: { key: 'value' },
+						},
+					}),
+				});
+
+				expect(outcome.result).toEqual([
+					withPairedItem(0, wrapIntoJson({ val: { key: 'value' } })),
+				]);
+			});
+
+			test.each<[CodeExecutionMode]>([['runOnceForAllItems'], ['runOnceForEachItem']])(
+				"does not return static data if it hasn't been modified in %s",
+				async (mode) => {
+					const outcome = await execTaskWithParams({
+						task: newTaskWithSettings({
+							code: `
+								const staticData = $getWorkflowStaticData("global");
+								return { val: staticData };
+							`,
+							nodeMode: mode,
+						}),
+						taskData: newDataRequestResponse(inputItems.map(wrapIntoJson), {
+							staticData: {
+								global: { key: 'value' },
+							},
+						}),
+					});
+
+					expect(outcome.staticData).toBeUndefined();
+				},
+			);
+
+			test.each<[CodeExecutionMode]>([['runOnceForAllItems'], ['runOnceForEachItem']])(
+				'returns the updated static data in %s',
+				async (mode) => {
+					const outcome = await execTaskWithParams({
+						task: newTaskWithSettings({
+							code: `
+								const staticData = $getWorkflowStaticData("global");
+								staticData.newKey = 'newValue';
+								return { val: staticData };
+							`,
+							nodeMode: mode,
+						}),
+						taskData: newDataRequestResponse(inputItems.map(wrapIntoJson), {
+							staticData: {
+								global: { key: 'value' },
+								'node:OtherNode': { some: 'data' },
```

Comment:
Interesting that we use the node name instead of the ID as a suffix.

Reply:
Yeah I know. We are unfortunately using node names in a bunch of places when we should use IDs instead :/